catalog/plugin/catalog-plugin-metacardbackup/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/MetacardBackupPlugin.java

```diff
@@ -96,28 +88,28 @@
             return processRequest;
         }
 
-        if (StringUtils.isEmpty(outputDirectory)) {
+        if (CollectionUtils.isEmpty(storageBackupPlugins)) {
             throw new PluginExecutionException(
-                    "Unable to delete backup ingested metacard; no output directory specified.");
+                    "Unable to delete backup ingested metacard; no metacard backup storage provider configured.");
         }
 
         List<ProcessDeleteItem> processUpdateItems = processRequest.getProcessItems();
         for (ProcessDeleteItem processUpdateItem : processUpdateItems) {
             Metacard metacard = processUpdateItem.getMetacard();
-            deleteBackupIfPresent(metacard.getId());
+            try {
+                for (MetacardBackupStorageProvider storageProvider : storageBackupPlugins) {
+                    storageProvider.deleteData(metacard.getId());
+                }
+            } catch (IOException | MetacardBackupException e) {
+                LOGGER.debug("Unable to delete backed up metacard data for metacard: {}",
```

Comment:
If you have seven backup storage providers and fail on the fourth in the list, you'll have deleted data from three but not the other four. (Arbitrary and ridiculous number of providers.)

Would it be better to move the try/catch inside the loop and debug but continue so you do the best cleanup you can?